src/test/scala/is/hail/stats/StatsSuite.scala

```diff
@@ -62,4 +62,14 @@ class StatsSuite extends SparkSuite {
       for (j <- 0 to 1)
         assert(G(i, j) == G1(i, j))
   }
+
+  @Test def poissonTest() {
+    // compare with R
+    assert(D_==(dpois(5, 10), 0.03783327))
+    assert(qpois(0.3, 10) == 8)
+    assert(qpois(0.3, 10, lowerTail = false) == 12)
+    assert(D_==(ppois(5, 10), 0.06708596))
+    assert(D_==(ppois(5, 10, lowerTail = false), 0.932914))
+    assert(rpois(5) >= 0)
```

Comment:
This one doesn't make sense: `ppois(qpois(ppois(x))) = x`

The largest k where this is true is 17:
```scala
assert(ppois(17, 1) != 1)
assert(ppois(17, 1, lowerTail = false) < 1)
```